### ISSUE TITLE
Linux Partitions Are Not Detected

### ISSUE THREAD

**Issue by MarioG-X** [dakanji/RefindPlus#194]
### RefindPlus Version

v0.14.2.AA Release

### Device Type

Apple Mac

### Problem Description

Comparing the logs below, you can see that 14.1 finds 3 Linux partitions on the external drive, which do not show up on 14.2. The is on an external drive on a Mac Mini. However, the same thing happens on a 2012 iMac with the Linux partitions on the internal drive, so external vs internal is not the issue. I see in the logs they don't show up in the "Scan External" section.

The 2 Tumbleweed partitions are BTRFS, don't remember what Elementary is. Oddly, 14.2 does display a fourth Linux partition which is Gparted.

### Problem Point

After starting a loader or tool

### Affected Items

Linux OS Loader

### Debug Log

Missing on 14.2:
[24t10r5501.log](https://github.com/user-attachments/files/17331266/24t10r5501.log)
Found on 14.1:
[24t10s0850.log](https://github.com/user-attachments/files/17331267/24t10s0850.log)

**Comment by dakanji** [comment]
You did not load the filesystem drivers needed to "find" the instances.

**Comment by MarioG-X** [comment]
Just was about to tell you I found the issue and fixed it, then the power went out! Was scheduled and be down all day! I think I noticed the folder was renamed and didn’t finish something.

my bad

**Comment by dakanji** [comment]
See [Discussion Item 196](https://github.com/dakanji/RefindPlus/discussions/196)

**Comment by dakanji** [comment]
UPDATE By @MarioG-X 

Power back on.
I thought something was screwy/confusing. See the two logs below, the directory drivers_x64 is found in both, in fact the same refind directory and files except for the efi file. 0.14.1 will show the 2 missing Linux partitions but 0.14.2 hot fix  does not.
Both using the same config file. Happens on both computers the same way with the same files. Can't find any dumb mistakes I may have made so far.

Bad 14.2:
[24t11a0821.log](https://github.com/user-attachments/files/17335815/24t11a0821.log)
Good 14.1
[24t11a1633.log](https://github.com/user-attachments/files/17335816/24t11a1633.log)

**Comment by dakanji** [comment]
Please share the log output from the same Mac as originally used or connect the external drive to this one.

**Comment by MarioG-X** [comment]
Will get them on the 2018 but don’t understand why the 2012 logs don’t help, they’re clear as day, both find the driver directory but 14.2 missing Linux. Both machines work with 14.1.

Do you only want me to test with the 2018 Mini going forward? 2012 too old? I am hoping to replace the 2012 with a 2017 iMac as soon as apple releases something this year. I plan the new iMac to use external only for Linux like the 2018 Mini.

**Comment by MarioG-X** [comment]
From 2018 Mini, same error as 2012 iMac:
4.2 bad
[24t11r4243.log](https://github.com/user-attachments/files/17344913/24t11r4243.log)
4.1 good
[24t11r4913.log](https://github.com/user-attachments/files/17344914/24t11r4913.log)

**Comment by dakanji** [comment]
Is the volume called "Elementary" actually ElementaryOS or is it another Tumbleweed Instance that just happens to be called that?

BTW, I essentially requested you share log from Unit A which already has External Disk connected or from Unit B after connecting External Disk to this.
How have you concluded that there was any interest in which one of Unit A or B was used and completely missed that the interest was in the external disk?

**Comment by MarioG-X** [comment]
There is a volume on the Mini with and old ElementaryOS on it, that is not Tumbleweed, it is a Linux distro. It does not exist on the iMac. It was for testing and should have been erased.

External is only on the Mini, it cannot run on the iMac which has RP on the internal SSD. Sorry it may have not been obvious. I cannot provide you with testing the external with the iMac.

Don't know what unit A and B are. I have a Mini and iMac. Any new RP releases start testing on the iMac and if it works I test on the Mini, but for some reason it seems you don't seem to want me to test on the iMac first or at all, which is why ask I 2012 no longer supported by RP?

I'll can keep running 14.1 on the iMac, just thought you would like to know problems/differences when you publish a new release. Both Mini and iMac boot everything with 14.1 so I can stick with that.

**Comment by dakanji** [comment]
I will rephrase ...

BTW, I essentially requested you share log from `Unit A (Mac Mini)` which already has External Disk connected **OR** from `Unit B (iMac)` after connecting External Disk to this.
That is, it didn't matter whether you used the Mini, iMac or anything else for that matter as long as the External Disk was present (for that specific test request).

Basically, I wanted to see whether Elementary, which I had noticed was on the External Disk from previously, would behave differently to Tumbleweed to know where to start looking at in detail.
The reason I was interested in Elementary is that my unit does not have issues showing Linux instances and I wanted to know whether issue you had was likely linked to symlinks.

Anyway, looks like it didn't matter in your case so I need to figure out how to find the issue.
BTW, don't get distracted by the external disk angle now. It is no longer relevant here

**Comment by MarioG-X** [comment]
I thought about symlinks since gparted which shows up does not use symlinks. But TW and elementary does. I put a vmlinuz kernel file in Tumbleweed2. And tested to see if it would show up since it was not a symlink. It did not. The menu still does not show TW1. TW2, elemOS. RP Did not find the vmlinuz kernel in TW2 /boot.
[24t12a2158.log](https://github.com/user-attachments/files/17348252/24t12a2158.log)

**Comment by dakanji** [comment]
I have no issues with this on my side: [24t12a4735.log](https://github.com/user-attachments/files/17348317/24t12a4735.log)
What is clear in your case is that regardless of what the log says, the added filesystem drivers are not loaded.
Why they are loading on my unit and not on your units needs thinking about.

Perhaps try the FS driver files included in the release package.

**Comment by MarioG-X** [comment]
Tried 14.1, 14.2, and release (log below). All same. You don't have btrfs, wonder if that's an issue. Strange all drivers work with 14.1.
[24t12b0254.log](https://github.com/user-attachments/files/17348340/24t12b0254.log)

**Comment by dakanji** [comment]
Was always going to come down to bisecting to find where things break. Just needed to have a idea before starting.
Try this to start off: https://file.io/RJMb2i5JG3Ep

**Comment by MarioG-X** [comment]
Here is log, I made sure 14.2 drivers were in directory.
[24t12k4542.log](https://github.com/user-attachments/files/17349751/24t12k4542.log)

**Comment by MarioG-X** [comment]
I was going to copy the TW BTRFS to an EXT4 to see if it made a difference. However, the Elementary partition is EXT4 so no need to test TW on EXT4. BTRFS vs EXT4 made no difference. The Gparted partition which shows up is a FAT partition.

**Comment by dakanji** [comment]
Please try this: https://file.io/g3O1TtqZKZqS

**Comment by MarioG-X** [comment]
No luck yet, 329a log:
[24t12s1141.log](https://github.com/user-attachments/files/17350692/24t12s1141.log)

Strange these lines found for only one partition but does not show on menu, must be vars thing:
```
  2:9100   0:1535  U P D A T E   D E F A U L T   S E L E C T I O N
  3:0064   0:0963  INFO: Changed to Previous Selection:- 'Load Instance: Linux - boot\vmlinuz-6.10.9-1-default from Volume:- Tumbleweed1'

```

If it's left from last boot vars, it could be wrong since the last boot before I post here is always Sequoia since that's where I am logged on to GitHub.

**Comment by dakanji** [comment]
Check this one: https://file.io/jZyPefWEeI01

**Comment by MarioG-X** [comment]
That did it. All partitions now show up:
[24t12t1537.log](https://github.com/user-attachments/files/17350848/24t12t1537.log)

**Comment by dakanji** [comment]
Thanks. That was actually just to pinpoint the breaking commit.
Now need to work through that to ID the file.

Some of the builds will be REL files. Just confirm whether things work or not.
https://file.io/nqICDAbSmKX3

The previous selection is what was selected on the previous RP or rEFInd boot the RP run being logged.

**Comment by MarioG-X** [comment]
330a does NOT show the Linux partitions.

**Comment by dakanji** [comment]
Next one: https://file.io/5KCMuwaLu0ee

**Comment by MarioG-X** [comment]
330b has all partitions showing, AOK.

**Comment by dakanji** [comment]
Ok. Please check this: https://file.io/ubiwnfzKZPIX

**Comment by MarioG-X** [comment]
332b2 does NOT show the 3 partitions. Bad.

**Comment by dakanji** [comment]
Thanks. Next one: https://file.io/5LZ8fOIxfd6V

**Comment by MarioG-X** [comment]
330b3 has all partitions showing, GOOD.

**Comment by MarioG-X** [comment]
Been standing by. Going to lunch should be back in a hour+.

**Comment by dakanji** [comment]
Thanks. Will pick things up tomorrow

**Comment by dakanji** [comment]
Please try this: https://file.io/yeUB0a64Tind

**Comment by MarioG-X** [comment]
330c has all partitions showing, GOOD.
Note there is new a Tumbleweed3 I am working on unrelated issue.

**Comment by dakanji** [comment]
Next One: https://file.io/teW79W58ZY9j
Will pick up in a few hours

**Comment by MarioG-X** [comment]
330d has all partitions showing, GOOD.

**Comment by dakanji** [comment]
Sorry. Might be a bit sporadic for a day or two.
Please try this next: https://file.io/EJVVvQvPGOlG

**Comment by MarioG-X** [comment]
330e has all partitions showing, GOOD.

> Sorry. Might be a bit sporadic for a day or two.

No problem

**Comment by dakanji** [comment]
Thanks. Please try this: https://temp-file-share.web.app/d/SYCH-t0if

**Comment by MarioG-X** [comment]
I was not even comfortable using file.io, but now another site? I prefer downloads from GitHub. Have been in computer security 40 years and extremely security conscience.

**Comment by dakanji** [comment]
Noted but not sure what difference the site that I put something for you to download from makes to your security.

I think the most relevant item security wise is the fact that I am providing you a file to run and not where it is put to download from. That is, you shouldn't be running the efi file at all if you have security concerns. Whether I add it here or elsewhere is actually irrelevant.

I add them to temporary holding sites because adding several files here becomes an issue. 

I do appreciate the security concern and probably best that you do not run such provided files at all as that is the real security issue.

Thanks for the help so far. I will look into the remaining items and make changes if I spot anything ... which incidentally, I don't think will be the case.

**Comment by MarioG-X** [comment]
Too bad GitHub has restrictions. I was ok with using one file site (file.io). Yes EFIs could be dangerous which is why I was more comfortable getting them from your GitHub.

**Comment by MarioG-X** [comment]
Something I think you should be aware of, and **happy** about, I am!:
I had to do some work involving installing the latest Tumbleweed on the 2012 iMac multiple times, which I had done before with RP. I had one of your recent test RP running at the time, when I booted the non-debug version and tried to boot the USB stick installer, RP showed the first line, then next line had something about "LoadedImageProtocol", familiar?

I know I had installed TW/USB with RP but it was quite some time ago so I went back to 14.1 - NOgo, then 14.0, Nogo! Wow, then I put on 330e and it all worked. Keep in mind I did not install TW for a while so could have been before 14.0. The good news is 330e worked fine. Thought this may be helpful.

**Comment by dakanji** [comment]
Took a look at what is left and nothing jumps out as being wrong.
Do not have the issue myself as said earlier so will leave things as they are.

Thanks.

**Comment by MarioG-X** [comment]
Thank you for all your hard work! I saw a lot of changes in the code and thought it would be a good idea to test everything out asap. Good thing. Anyway, I will do a full test after you release the fixed code and report back. I am sure it will be fine since 330e has no issues.

As usual, I will be glad to do any pre-release testing you would like to do to prevent issues.

**Comment by dakanji** [comment]
Thanks but perhaps it wasn't clear that I couldn't see any issues in the code that was yet to be tested as at where the test runs stopped and I do not have issues on my unit. Hence, no changes are being made at this time. Will revisit if anything pops up. 330e is not much different to 0.14.1.

**Comment by MarioG-X** [comment]
OK, well, I can use 14.1 for everything except installing TW via a USB stick. And  unlike 14.2, 14.1 debug works but probably never need. I can use 330e for everything including TW installation, except no debugging efi. Hopefully, someone else will report these errors as they upgrade so they get addressed permanently. Problem is, once RP works there is little need to upgrade, I only upgraded to provide/report feedback to you. Until next release, thanks again.

**Comment by dakanji** [comment]
> You don't have btrfs, wonder if that's an issue

You might have been on to something here.
Can you try the btrfs driver from v0.13.3.AD when you are able to?

**Comment by MarioG-X** [comment]
I thought nothing would change because this also affected the Elementary Linux (ext4) which would not show up, so not btrfs as I first thought. Here are results:

14.2 and 14.2 hot fix with respective btrfs drivers did not work. Test was to make sure problem still exists.
14.2 hot fix with btrfs driver from 13.3AD did not work.

**Comment by dakanji** [comment]
Yeah. The Ext4 angle did suggest it was not this. Still needed to exclude changes that went into the driver.

**Comment by dakanji** [comment]
Likely related to this: https://sf.net/p/refind/discussion/general/thread/172b9cbf09/

**Comment by dakanji** [comment]
Continued in Discussion https://github.com/RefindPlusRepo/RefindPlus/discussions/242
